Covid19Radar/Covid19Radar/Services/ExposureDetectionService.cs

```diff
@@ -0,0 +1,156 @@
+    }
+
+    public class ExposureDetectionService : IExposureDetectionService
+    {
+        private readonly ILoggerService _loggerService;
+        private readonly IUserDataRepository _userDataRepository;
+        private readonly ILocalNotificationService _localNotificationService;
+        private readonly IExposureRiskCalculationService _exposureRiskCalculationService;
+
+        private readonly IExposureConfigurationRepository _exposureConfigurationRepository;
+
+        private readonly IExposureDataCollectServer _exposureDataCollectServer;
+
+        public ExposureDetectionService
+        (
+            ILoggerService loggerService,
+            IUserDataRepository userDataRepository,
+            ILocalNotificationService localNotificationService,
+            IExposureRiskCalculationService exposureRiskCalculationService,
+            IExposureConfigurationRepository exposureConfigurationRepository,
+            IExposureDataCollectServer exposureDataCollectServer
+            )
+        {
+            _loggerService = loggerService;
+            _userDataRepository = userDataRepository;
+            _localNotificationService = localNotificationService;
+            _exposureRiskCalculationService = exposureRiskCalculationService;
+            _exposureConfigurationRepository = exposureConfigurationRepository;
+            _exposureDataCollectServer = exposureDataCollectServer;
+        }
+
+        public void DiagnosisKeysDataMappingApplied()
+        {
+            _loggerService.StartMethod();
+
+            if (_exposureConfigurationRepository.IsDiagnosisKeysDataMappingConfigurationUpdated())
+            {
+                _exposureConfigurationRepository.SetDiagnosisKeysDataMappingAppliedDateTime(DateTime.UtcNow);
+                _exposureConfigurationRepository.SetIsDiagnosisKeysDataMappingConfigurationUpdated(false);
+            }
+
+            _loggerService.EndMethod();
+        }
+
+        public void PreExposureDetected(ExposureConfiguration exposureConfiguration, string enVersion)
+        {
+            _loggerService.Debug("PreExposureDetected");
+        }
+
+        public void ExposureDetected(ExposureConfiguration exposureConfiguration, string enVersion, IList<DailySummary> dailySummaries, IList<ExposureWindow> exposureWindows)
+        {
+            _loggerService.Debug("ExposureDetected: ExposureWindows");
+
+            _ = Task.Run(async () =>
+            {
+                await _userDataRepository.SetExposureDataAsync(
+                    dailySummaries.ToList(),
+                    exposureWindows.ToList()
+                    );
+
+                bool isHighRiskExposureDetected = dailySummaries
+                    .Select(dailySummary => _exposureRiskCalculationService.CalcRiskLevel(dailySummary))
+                    .Where(riskLevel => riskLevel >= RiskLevel.High)
+                    .Count() > 0;
```

Comment:
[修正任意]
このパターンは `.Any()` が使えます。

```suggestion
                    .Any(riskLevel => riskLevel >= RiskLevel.High)
```

Reply:
eb81000 で対応しました！